10up/gutenbridge
# Added Settings UI

Closes #38, #54

### Description of the Change
This PR adds UI for Plugin settings to configure the `post_types` supported by the plugin. Currently, it is being done with the help of a filter hook. With this change, users will be able to visually select the `post_types` they wish to be configured with the plugin.

Also, missing [yoast/phpunit-polyfills](https://packagist.org/packages/yoast/phpunit-polyfills) package has been added to enable running of unit tests.

**To-do:**

- [x] Polishing PR
- [x] Configure filter to use `post_types` value from DB to configure the plugin

### Possible Drawbacks

None that I can think of.

### Verification Process



To check the working of the Settings panel, follow the steps below:

- Click "Convert to Blocks" under the "Settings" menu.
- On the **Settings** page, de-select one of the default post types, i.e. either page or post.
- Navigate to the de-selected item's menu (Page, Post or CPT) and verify that it is no longer supported by the plugin.

### Checklist:






### Changelog Entry

> Added - Settings UI for managing supported **post_types**
